src/fondant/cli.py

```diff
@@ -360,3 +393,54 @@ def pipeline_from_string(import_string: str) -> Pipeline:
         raise ImportFromStringError(msg)
 
     return instance
+
+
+def component_from_module(module_str: str) -> t.Type[Component]:
+    """Try to import a component from a module otherwise raise an ImportFromModuleError."""
+    if ".py" in module_str:
+        module_str = module_str.rsplit(".py", 1)[0]
+
+    module_str = module_str.replace("/", ".")
+
+    try:
+        class_members = inspect.getmembers(
+            importlib.import_module(module_str),
+            inspect.isclass,
+        )
+    except ModuleNotFoundError:
+        msg = f"`{module_str}` was not found. Please provide a valid module."
+        raise ImportFromModuleError(
+            msg,
+        )
+
+    # Define the order in the method resolution order (MRO) to identify the level of inheritance
+    # depth. This corresponds to the user-defined component class that subclasses specified load/
+    # transform component (e.g. DaskLoadComponent, PandasTransformComponent, ...)
+    base_component_order = 2
+    component = None
+
+    for name, cls in class_members:
+        method_order = cls.__mro__
+
+        # Check if the MRO has a sufficient depth to identify the user-defined component class,
+        # and if the class at that depth is the BaseComponent class.
+        if (
+            len(method_order) > base_component_order
+            and method_order[base_component_order] == BaseComponent
+        ):
+            logger.info(f"Component `{name}` found in module {module_str}")
+
+            if component is not None:
+                msg = (
+                    f"More than one component found in module {module_str}. Only one component "
+                    f"can be present"
+                )
+                raise ImportFromModuleError(msg)
+
+            component = cls
+
+    if component is None:
+        msg = f"No Component found in module {module_str}"
+        raise ImportFromModuleError(msg)
+
+    return component
```

Comment:
This will break if the user creates any base component classes to inherit from in their own project. We can probably make this a bit more robust by taking the component class with the max order. Something like this (untested):

```python
    component_classes = defaultdict(list)

    for name, cls in class_members:
        if issubclass(cls, BaseComponent):
            order = len(cls.__mro__)
            component_classes[order].append(cls)
            
    max_order = max(component_classes)
    found_components = component_classes[max_order]
    
    if len(found_components) == 0:
        msg = f"No Component found in module {module_str}"
        raise ImportFromModuleError(msg)
    if len(found_components) > 1:
        msg = (
            f"Found multiple components in {module_str}: {found_components}. Only one component "
            f"can be present"
        )
        raise ImportFromModuleError(msg)

    logger.info(f"Component `{name}` found in module {module_str}")
    return found_components[0]
```

Reply:
Tried it out with small tweaks and it works